### PR TITLE
Added slices actions and reducers

### PR DESCRIPTION
**Pull request by Asiimwe-Mark**
Added Books and Categories slices, reducers, and actions following the requirements below;

Add Redux Toolkit (npm install react-redux @reduxjs/toolkit).
Create a directory that will contain all your Redux logic (/src/redux)
Configure a Redux store (/src/redux/store.js)
Define a slice of state for books that:
Includes an array of books (initial state: empty array)
Includes a reducer that adds a book
Includes a reducer that removes a book
Define a slice of state for categories that:
Includes an array of categories (initial state: empty array)
Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

### CODE REVIEW

File: src/feature/redux/store.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (julie-ify) [inline src/feature/redux/store.js]: - [ ] Kindly avoid disabling linters. They are necessary for upholding coding best practices and uniformity.

File: src/feature/redux/books/bookSlice.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (julie-ify) [inline src/feature/redux/books/bookSlice.js]: - [ ] Kindly avoid disabling linters. They are necessary for upholding coding best practices and uniformity.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @As1imwe-Mark,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @As1imwe-Mark, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**